java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiColumnValue.java

```diff
@@ -134,12 +135,22 @@ public void unpackMap(List<ColumnValue> keys, List<ColumnValue> values) {
     public void unpackStruct(List<Integer> structFieldIndex, List<ColumnValue> values) {
         StructObjectInspector inspector = (StructObjectInspector) fieldInspector;
         List<? extends StructField> fields = inspector.getAllStructFieldRefs();
+        // if we don't do any prune, we preserve all fields.
+        if (structFieldIndex == null) {
+            structFieldIndex = new ArrayList<>(fields.size());
+            for (int i = 0; i < fields.size(); i++) {
+                structFieldIndex.add(i);
+            }
+        }
         for (int i = 0; i < structFieldIndex.size(); i++) {
-            StructField sf = fields.get(structFieldIndex.get(i));
-            Object o = inspector.getStructFieldData(fieldData, sf);
+            Integer idx = structFieldIndex.get(i);
             HudiColumnValue cv = null;
-            if (o != null) {
-                cv = new HudiColumnValue(sf.getFieldObjectInspector(), o);
+            if (idx != null) {
```

Comment:
when is `idx` NULL

Reply:
When field is not found. For example,  BE thinks struct is `<B:int, a:int>`, but actually in hudi it's `<b:int, a:int>`'